rpc/src/rpc.rs

```diff
@@ -1323,11 +1417,20 @@ impl JsonRpcRequestProcessor {
 
         // Finalized blocks
         let mut blocks: Vec<_> = self
-            .blockstore
-            .rooted_slot_iterator(max(start_slot, lowest_blockstore_slot))
-            .map_err(|_| Error::internal_error())?
-            .filter(|&slot| slot <= end_slot && slot <= highest_super_majority_root)
-            .collect();
+            .runtime
+            .spawn_blocking({
+                let blockstore = Arc::clone(&self.blockstore);
+                move || -> Result<Vec<_>> {
+                    let blocks = blockstore
+                        .rooted_slot_iterator(max(start_slot, lowest_blockstore_slot))
```

Comment:
same here: I think that this iterator is fast, it just gets a bunch of u64s from
rocks.

In general not sure that this function should be async, it has a confusing name
but it just returns a vec of slot numbers